File: code/typescript/src/mediator/ScrumMaster.ts
import { Colleague } from "./Colleague";
import { Mediator } from "./Mediator";
import { Developer } from "./Developer";
import { Tester } from "./Tester";

export class ScrumMaster implements Mediator {
    private developer!: Developer;
    private tester!: Tester;

    setDeveloper(developer: Developer): void {
        this.developer = developer;
    }

    setTester(tester: Tester): void {
        this.tester = tester;
    }

    request(message: string, colleague: Colleague): void {
        if (colleague instanceof Developer) {
            this.tester.receive(message);
        } else if (colleague instanceof Tester) {
            this.developer.receive(message);
        }
    }

    response(message: string, colleague: Colleague): void {
        console.log(`Scrum Master handling response: ${message}`);
    }
}

File: code/python/mediator/ScrumMaster.py
from Mediator import Mediator


class ScrumMaster(Mediator):
    def __init__(self):
        self.developer = None
        self.tester = None

    def set_developer(self, developer):
        self.developer = developer

    def set_tester(self, tester):
        self.tester = tester

    def request(self, message, colleague):
        if colleague == self.developer:
            self.tester.receive(message)
        elif colleague == self.tester:
            self.developer.receive(message)

    def response(self, message, colleague):
        print(f"Scrum Master handling response: {message}")
